### DISCUSSION TITLE
All Restarts Require New Job-orders. How to Properly Restart without Overload.

### DISCUSSION THREAD

**Discussion by Cminer3207** [swar/Swar-Chia-Plot-Manager#392] [General]
It appears that any restart, with or without drive changes, will sometimes cause the manager to complete current plots and not start any new ones. 

Swar tells us 
For restarts with drive changes, there will be path problems... so... 
For currently running job-orders:
Set max_plots: 0
(This will finish all currently running jobs, without starting more. It appears to do this anyway).

Create new job-orders to replace the existing job-orders we just cancelled. 

Copy & Paste Original Job-orders. 
Rename them. 
Set max_plots: to whatever you had before (minus the jobs that are still finishing under the original job-orders) 

It is now obviously the manager needs new job-orders for ANY restart, or it will just complete the current temp plots and then stop. 

Problem:
If your manager isn’t already at max_concurrent (because it isn’t, or because you increased it during the restart) it will ignore max_phase1: limits and immediately begin new staggered plots until it reaches max_concurrency. 
You will twice as many Phase 1’s running as your max. 
This will disrupt your stagger until the Original Job-order plots are done, almost certainly overloading the CPU in phase1 and then overloading temp drive in Phase 2-4. 

SOLUTION: 
OPTION1: For restarts when Manager IS at max_concurrency:
Create New job-order by copy & paste Original job-order. 
Give it a New job-order a different name. 
python manager.py restart  
(Wait 1-2 minutes)
python manager.py view
(check that plots are progressing). 
Done. 

OPTION2: For restarts when Manager IS NOT at max_concurrency: 
Do not create New job-order... yet. 
Restart manager. 
Wait until all currently running plots have exited Phase 1!!!
Then execute restart as we did above for OPTION1. 

If you don’t create a New job-order(s), the Manager will finish current orders and then stop. 

If you do create New job-orders but don’t double-restart, and max_concurrency allows the Manager to create new plots, the new plots will disrupt your stagger and eventually overload your CPU or temp space at some point.

**Comment by Cminer3207** [comment]
RESTART WITHOUT OVERLOAD: SIMPLE:

Manager is AT max_concurrecy:
Set current job-order max_plots: 0
Create new Job-order(s) and restart.

Manager is NOT AT max_concurrency:
Set current job-order max_plots: 0
Restart WITHOUT New job-order(s).
Wait until all plots are beyond Phase1. Create New job order(s) and restart again.

You can probably skip the “max_plots: 0” part because any restart will automatically stop creating new plots your the original job(s).

**Reply by ryanng3120** [reply]
hey thanks for sharing this, would following the above steps delete the temp files from the incomplete plots, i.e. from the previous job order where max_plots is zero?
